SAP/spartacus
# fix: CXSPA-2919 Issue with false consent state

closes [CXSPA-2919](https://jira.tools.sap/browse/CXSPA-2919)

![image](https://user-images.githubusercontent.com/28891782/229783414-9dc45b28-b56a-4943-8213-e049cfb616a9.png)

The following PR contains a solution to the incorrect user consent state in the "Consent management" view. The problem occurred when CDS was active due to excessive HTTP calls per user consent. The excess and uncontrolled amount of HTTP calls were caused by the `userConsentService.getConsent()` method calling the `getConsents(true)` method inside a `switchMap` to retrieve consent if it is not in the store. This method listened to every change in user consent in the state, causing the HTTP requests to retrieve consent to occur an uncontrolled number of times and with incorrect parameters (resulting in anonymous consent being saved in the user consent section) as long as there was at least one subscriber (and there are many due to the profile tag service provided by the feature-lib CDS).
To avoid such a scenario, the `getConsents()` method has been extended with operator `take` in a place where logic was responsible for calling a method that will load consents if needed.

---
Steps to test:

1. enable CDS
2. login
3. go to consent management and change some consent
4. logout and log in again as the same user
5. go to the consent management view and check whether the view contains changes applied in step 3.
